SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -624,6 +646,35 @@ void USpatialSender::TrackRPC(AActor* Actor, UFunction* Function, const RPCPaylo
 }
 #endif
 
+bool USpatialSender::WillHaveAuthorityOverActor(AActor* TargetActor, Worker_EntityId TargetEntity)
+{
+	bool WillHaveAuthorityOverActor = true;
+
+	if (GetDefault<USpatialGDKSettings>()->bEnableOffloading)
+	{
+		if (!USpatialStatics::IsActorGroupOwnerForActor(TargetActor))
+		{
+			WillHaveAuthorityOverActor = false;
+		}
+	}
+
+	if (GetDefault<USpatialGDKSettings>()->bEnableUnrealLoadBalancer)
+	{
+		if (NetDriver->VirtualWorkerTranslator != nullptr)
+		{
+			if (const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(TargetEntity))
+			{
+				if (AuthorityIntentComponent->VirtualWorkerId != NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId())
```

Comment:
I wonder if this check might be too aggressive - perhaps this is set correctly, but the local worker hasn't processed the state change yet. Thoughts?

Reply:
If we've sent this RPC _before_ processing the state change, doesn't that mean the code would have been in a context of no authority? In which case it shouldn't have expected to be able to send the RPC?

This should catch the main case of firing RPCs on actors we've just spawned but not gained authority of yet.

Reply:
Should this be resolved?

Reply:
It checks the authority of the specific RPC component, but I don't have a good use-case for when this permutation could occur. Something we should watch out for though.